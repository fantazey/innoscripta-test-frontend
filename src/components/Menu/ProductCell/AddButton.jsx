import React from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';
import CurrencyPrice from '../../CurrencyPrice';

const ProductCellAddButton = ({ add, price }) => {
  const { t } = useTranslation();
  return (
    <div
      className="d-flex btn btn-info align-self-stretch justify-content-start"
      onClick={add}
    >
      {t('add-to-cart')}
      <span className="px-2 align-right">
        <CurrencyPrice price={price} />
      </span>
    </div>
  );
};

ProductCellAddButton.propTypes = {
  add: PropTypes.func.isRequired,
  price: PropTypes.number.isRequired
};

export default ProductCellAddButton;
