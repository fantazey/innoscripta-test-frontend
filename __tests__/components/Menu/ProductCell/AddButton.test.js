import AddButton from '../../../../src/components/Menu/ProductCell/AddButton';

describe('Menu/ProductCell/AddButton', () => {
  let wrapper;
  const defaultProps = {
    add: jest.fn(),
    price: 12.12
  };

  beforeEach(() => {
    wrapper = shallow(<AddButton {...defaultProps} />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle click', () => {
    wrapper.simulate('click');
    expect(defaultProps.add).toHaveBeenCalledTimes(1);
  });
});
