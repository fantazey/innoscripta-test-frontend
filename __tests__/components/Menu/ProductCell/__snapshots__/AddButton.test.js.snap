// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu/ProductCell/AddButton should match snapshot 1`] = `
<div
  className="d-flex btn btn-info align-self-stretch justify-content-start"
  onClick={[MockFunction]}
>
  _add-to-cart
  <span
    className="px-2 align-right"
  >
    <CurrencyPrice
      price={12.12}
    />
  </span>
</div>
`;
